core/domain/prod_validation_jobs_one_off.py

```diff
@@ -5038,6 +5039,119 @@ def _get_custom_validation_functions(cls):
         ]
 
 
+class PlaythroughModelValidator(BaseModelValidator):
+    """Class for validating PlaythroughModel."""
+
+    PLAYTHROUGH_PROJECT_RELEASE_DATETIME = datetime.datetime(2018, 9, 1)
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'exp_ids': (
+                exp_models.ExplorationModel, [item.exp_id]),
+        }
+
+    @classmethod
+    def _get_model_id_regex(cls, unused_item):
+        return r'^[A-Za-z0-9-_]{1,%s}\.[A-Za-z0-9-_]{1,%s}$' % (
+            base_models.ID_LENGTH, base_models.ID_LENGTH)
+
+    @classmethod
+    def _get_model_domain_object_instance(cls, item):
+        return stats_services.get_playthrough_from_model(item)
+
+    @classmethod
+    def _validate_exploration_id_in_whitelist(cls, item):
+        """Validate the exploration id in playthrough model is in
+        the whitelist.
+
+        Args:
+            item: ndb.Model. PlaythroughModel to validate.
+        """
+        whitelisted_exp_ids_for_playthroughs = (
+            config_domain.WHITELISTED_EXPLORATION_IDS_FOR_PLAYTHROUGHS.value)
+
+        if item.exp_id not in whitelisted_exp_ids_for_playthroughs:
+            cls.errors['exploration id check'].append(
+                'Entity id %s: recorded in exploration_id:%s which '
+                'has not been curated for recording.' % (
+                    item.id, item.exp_id)
+            )
+
+    @classmethod
+    def _validate_reference(cls, item):
+        """Validate the playthrough is referenced by some issues of
+        the corresponding exploration.
+
+        Args:
+            item: ndb.Model. PlaythroughModel to validate.
+        """
+        exp_id = item.exp_id
+        exp_version = item.exp_version
+        exp_issues = (
+            stats_models.ExplorationIssuesModel.get_model(exp_id, exp_version))
```

Comment:
Why aren't we handling this through `_get_external_id_relationships`?

Reply:
Done, added an entry in `_get_external_id_relationships`.